src/Statistics.jl

```diff
@@ -286,12 +292,12 @@ function median!(
     x0=x[1],
     stop_iter=1000000,
     kwargs...
-) where {T}
-    x0 === nothing || copyto!(y, x0)
+)
     n = length(x)
+    (length(w) != n) && throw(DimensionMismatch("The number of weights ($(length(w))) does not match the number of points for the median ($(n))."))
+    x0 === nothing || copyto!(y, x0)
```

Comment:
I think we discussed earlier, that `y` might carry the initial value, but I don't see that documented here... Maybe with having `x0` anyways there is no need to assume that, it's either `x[1]` or set by the user to a proper initial value, so we should always copy it to `y` here I think? Is there a reason to actually actively pass `nothing`?

Reply:
It's documented here https://github.com/JuliaNLSolvers/Manifolds.jl/blob/6fdb918ff35edfb00573a4edec74fd819ce27051/src/Statistics.jl#L69-L74

The use case for `x0=nothing` is when `y` already contains an initial value that you want to refine in-place without allocating a new vector. This isn't performance critical, so if this is confusing, we can change it.

Alternatively, we could do `y === x0 || copyto!(y, x0)`. Then for the use case I'm thinking of, one would call `mean!(M, y, x, w; x0=y)`, which would also avoid the allocation and copy.

Reply:
Hm, oh ;) That was me. I think your alternative is really less confusing, so I would opt for that.